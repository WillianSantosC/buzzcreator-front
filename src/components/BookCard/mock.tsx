export const items = [
  {
    id: 1,
    titulo: "Bíblia Slim Filament Marrom - NVT",
    autor: "Mundo Cristão",
    imagem:
      "https://images.tcdn.com.br/img/img_prod/400550/pre_venda_biblia_slim_filament_preta_nvt_2061_2_af82f09586aacb524dfa755e81ce3ec3.jpg",
    preco: 165,
    descricao:
      "A Bíblia Filament é a versão mais inovadora que já foi publicada em língua portuguesa. Ela une a simplicidade e clareza da Nova Versão Transformadora (NVT) com uma experiência digital imersiva que vai surpreender você! Além do texto impresso, a Bíblia oferece acesso gratuito ao aplicativo Filament, que é fácil de utilizar e expande sua leitura com recursos incríveis, diretamente no seu celular ou tablet.",
  },
  {
    id: 2,
    titulo: "Bíblia Slim Filament Preta - NVT",
    autor: "Mundo Cristão",
    imagem:
      "https://images.tcdn.com.br/img/img_prod/400550/biblia_365_naa_edicao_limitada_2071_1_13e6ec29c90c9550d5de5d647f6f3c43.jpg",
    preco: 165,
    descricao:
      "A Bíblia Filament é a versão mais inovadora que já foi publicada em língua portuguesa. Ela une a simplicidade e clareza da Nova Versão Transformadora (NVT) com uma experiência digital imersiva que vai surpreender você! Além do texto impresso, a Bíblia oferece acesso gratuito ao aplicativo Filament, que é fácil de utilizar e expande sua leitura com recursos incríveis, diretamente no seu celular ou tablet.",
  },
  {
    id: 3,
    titulo: "Bíblia 365 NAA - Edição Limitada",
    autor: "Mundo Cristão",
    imagem:
      "https://images.tcdn.com.br/img/img_prod/400550/pre_venda_biblia_slim_filament_marrom_nvt_2059_2_6b31a81586efbab2051f46e7cb748571.jpg",
    preco: 70,
    descricao:
      "A Bíblia 365 chegou em uma versão exclusiva e ainda mais especial! Além de oferecer um plano de leitura diário cuidadosamente elaborado, que combina porções do Antigo Testamento, Novo Testamento, Salmos e Provérbios, esta edição traz um toque único: uma jaqueta personalizada que transforma o produto em uma verdadeira obra de arte.",
  },
  {
    id: 4,
    titulo: "Box das Crianças",
    autor: "JesusCopy",
    imagem:
      "https://images.tcdn.com.br/img/img_prod/400550/box_dia_das_criancas_2037_1_ef071c6ab942531f4bd19842274c3b97.jpg",
    preco: 90,
    descricao:
      "A Bíblia comentada Kids é o novo projeto do JesusCopy que nós desenvolvemos com o objetivo de dar ferramenta para os pais criarem uma paixão em seus filhos pela Bíblia. Ela contém 50 capítulos com 50 conteúdos e a cada conteúdo trazemos um versículo, uma reflexão, uma oração e uma atividade lúdica para que você possa fazer junto com seu filho.",
  },
  {
    id: 5,
    titulo: "Bíblia Slim Filament Marrom - NVT",
    autor: "Mundo Cristão",
    imagem:
      "https://images.tcdn.com.br/img/img_prod/400550/pre_venda_biblia_slim_filament_preta_nvt_2061_2_af82f09586aacb524dfa755e81ce3ec3.jpg",
    preco: 165,
    descricao:
      "A Bíblia Filament é a versão mais inovadora que já foi publicada em língua portuguesa. Ela une a simplicidade e clareza da Nova Versão Transformadora (NVT) com uma experiência digital imersiva que vai surpreender você! Além do texto impresso, a Bíblia oferece acesso gratuito ao aplicativo Filament, que é fácil de utilizar e expande sua leitura com recursos incríveis, diretamente no seu celular ou tablet.",
  },
  {
    id: 6,
    titulo: "Bíblia Slim Filament Preta - NVT",
    autor: "Mundo Cristão",
    imagem:
      "https://images.tcdn.com.br/img/img_prod/400550/biblia_365_naa_edicao_limitada_2071_1_13e6ec29c90c9550d5de5d647f6f3c43.jpg",
    preco: 165,
    descricao:
      "A Bíblia Filament é a versão mais inovadora que já foi publicada em língua portuguesa. Ela une a simplicidade e clareza da Nova Versão Transformadora (NVT) com uma experiência digital imersiva que vai surpreender você! Além do texto impresso, a Bíblia oferece acesso gratuito ao aplicativo Filament, que é fácil de utilizar e expande sua leitura com recursos incríveis, diretamente no seu celular ou tablet.",
  },
  {
    id: 7,
    titulo: "Bíblia 365 NAA - Edição Limitada",
    autor: "Mundo Cristão",
    imagem:
      "https://images.tcdn.com.br/img/img_prod/400550/pre_venda_biblia_slim_filament_marrom_nvt_2059_2_6b31a81586efbab2051f46e7cb748571.jpg",
    preco: 70,
    descricao:
      "A Bíblia 365 chegou em uma versão exclusiva e ainda mais especial! Além de oferecer um plano de leitura diário cuidadosamente elaborado, que combina porções do Antigo Testamento, Novo Testamento, Salmos e Provérbios, esta edição traz um toque único: uma jaqueta personalizada que transforma o produto em uma verdadeira obra de arte.",
  },
  {
    id: 8,
    titulo: "Box das Crianças",
    autor: "JesusCopy",
    imagem:
      "https://images.tcdn.com.br/img/img_prod/400550/box_dia_das_criancas_2037_1_ef071c6ab942531f4bd19842274c3b97.jpg",
    preco: 90,
    descricao:
      "A Bíblia comentada Kids é o novo projeto do JesusCopy que nós desenvolvemos com o objetivo de dar ferramenta para os pais criarem uma paixão em seus filhos pela Bíblia. Ela contém 50 capítulos com 50 conteúdos e a cada conteúdo trazemos um versículo, uma reflexão, uma oração e uma atividade lúdica para que você possa fazer junto com seu filho.",
  },
];
