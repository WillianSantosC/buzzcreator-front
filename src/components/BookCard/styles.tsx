import { styled } from "../../../styled-system/jsx";

export const Wrapper = styled("article", {
  base: {
    position: "relative",
    display: "flex",
    flexDirection: "column",
    width: "100%",
    height: "100%",
    backgroundColor: "token(colors.white)",
  },
});

export const ImageBox = styled("div", {
  base: {
    height: "20rem",
    width: "100%",
    background: "#f6f7f8",
    backgroundImage:
      "linear-gradient( to right,#f6f7f8 0%,#edeef1 20%,#f6f7f8 40%,#f6f7f8 100%)",
    backgroundSize: "80rem 14rem",
    animation: "placeholderShimmer 1s linear infinite forwards",

    "& > img": {
      width: "100%",
      height: "100%",
      objectFit: "cover",
    },
  },
});

export const Content = styled("div", {
  base: {
    display: "flex",
    flexDirection: "column",
    justifyContent: "space-between",
    position: "relative",
    height: "100%",
    margin: "token(spacings.xsmall)",
  },
});

export const Info = styled("div", {
  base: {
    maxWidth: "calc(100% - 2.5rem)",
  },
});

export const Title = styled("h3", {
  base: {
    fontSize: "token(font.sizes.medium)",
    lineHeight: "token(font.sizes.medium)",
    fontWeight: "token(font.bold)",
    color: "token(colors.black)",
  },
});

export const Author = styled("h4", {
  base: {
    fontSize: "token(font.sizes.small)",
    fontWeight: "token(font.bold)",
    color: "token(colors.gray)",
  },
});

export const BuyBox = styled("div", {
  base: {
    display: "flex",
    alignItems: "center",
    justifyContent: "flex-end",
    marginTop: "token(spacings.xxsmall)",
  },
});

export const Price = styled("div", {
  base: {
    display: "inline-flex",
    fontWeight: "token(font.bold)",
    height: "3rem",
    alignItems: "center",
    color: "token(colors.white)",
    padding: "0 token(spacings.xxsmall)",
    backgroundColor: "token(colors.secondary)",
    borderRadius: "token(border.radius)",
    marginRight: "calc(token(spacings.xxsmall) / 2)",
  },
});
