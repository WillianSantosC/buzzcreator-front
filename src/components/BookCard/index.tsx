import * as S from "./styles";

export type BookCardProps = {
  id: number;
  titulo: string;
  autor: string;
  imagem: string;
  preco: number;
  descricao: string;
  onClick?: () => void;
};

const BookCard = ({ titulo, autor, imagem, preco, onClick }: BookCardProps) => (
  <S.Wrapper onClick={onClick}>
    <S.ImageBox>
      <img src={imagem} alt={titulo} />
    </S.ImageBox>
    <S.Content>
      <S.Info>
        <S.Title>{titulo}</S.Title>
        <S.Author>{autor}</S.Author>
      </S.Info>
      <S.BuyBox>
        <S.Price>
          {preco.toLocaleString("pt-BR", {
            style: "currency",
            currency: "BRL",
          })}
        </S.Price>
      </S.BuyBox>
    </S.Content>
  </S.Wrapper>
);
export default BookCard;
