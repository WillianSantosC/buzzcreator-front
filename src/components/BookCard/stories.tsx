import { Meta, StoryObj } from "@storybook/react";

import BookCard, { BookCardProps } from ".";

export default {
  title: "components/BookCard",
  component: BookCard,
  parameters: {
    layout: "fullscreen",
    backgrounds: {
      default: "won-dark",
    },
  },
  args: {
    titulo: "Box das Crian√ßas",
    autor: "JesusCopy",
    imagem:
      "https://images.tcdn.com.br/img/img_prod/400550/box_dia_das_criancas_2037_1_ef071c6ab942531f4bd19842274c3b97.jpg",
    preco: 90,
  },
} as Meta;

export const Default: StoryObj<BookCardProps> = {
  render: (args) => (
    <div style={{ width: "30rem", margin: " 0 auto" }}>
      <BookCard {...args} />
    </div>
  ),
};
