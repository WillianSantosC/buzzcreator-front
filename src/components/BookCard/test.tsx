import "@testing-library/jest-dom";
import { fireEvent, render, screen } from "@testing-library/react";
import BookCard, { BookCardProps } from ".";

const props: BookCardProps = {
  id: 1,
  titulo: "Clean Code",
  autor: "Robert C. Martin",
  imagem: "/img/books.jpg",
  preco: 99.9,
  descricao: "Um livro sobre boas práticas de programação",
  estoque: 10,
  onClick: jest.fn(),
};

describe("<BookCard />", () => {
  it("should render the book title", () => {
    render(<BookCard {...props} />);
    expect(screen.getByText(/clean code/i)).toBeInTheDocument();
  });

  it("should render the book author", () => {
    render(<BookCard {...props} />);
    expect(screen.getByText(/robert c. martin/i)).toBeInTheDocument();
  });

  it("should render the book image with alt text", () => {
    render(<BookCard {...props} />);
    const image = screen.getByRole("img", { name: /clean code/i });
    expect(image).toHaveAttribute("src", props.imagem);
    expect(image).toHaveAttribute("alt", props.titulo);
  });

  it("should call onClick when clicked", () => {
    render(<BookCard {...props} />);
    const card = screen
      .getByRole("img", { name: /clean code/i })
      .closest("div")!;
    fireEvent.click(card);
    expect(props.onClick).toHaveBeenCalledTimes(1);
  });

  it("matches snapshot", () => {
    const { container } = render(<BookCard {...props} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
