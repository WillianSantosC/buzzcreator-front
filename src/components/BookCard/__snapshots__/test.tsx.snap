// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BookCard /> matches snapshot 1`] = `
<article
  class="pos_relative d_flex flex-d_column w_100% h_100% bg-c_token(colors.white) cursor_pointer"
>
  <div
    class="h_20rem w_100% bg_#f6f7f8 bg-i_linear-gradient(_to_right,#f6f7f8_0%,#edeef1_20%,#f6f7f8_40%,#f6f7f8_100%) bg-s_80rem_14rem anim_placeholderShimmer_1s_linear_infinite_forwards [&_>_img]:w_100% [&_>_img]:h_100% [&_>_img]:obj-f_cover"
  >
    <img
      alt="Clean Code"
      src="/img/books.jpg"
    />
  </div>
  <div
    class="d_flex flex-d_column jc_space-between pos_relative h_100% m_token(spacings.xsmall)"
  >
    <div
      class="max-w_calc(100%_-_2.5rem)"
    >
      <h3
        class="fs_token(font.sizes.medium) lh_token(font.sizes.medium) fw_token(font.bold) c_token(colors.black)"
      >
        Clean Code
      </h3>
      <h4
        class="fs_token(font.sizes.small) fw_token(font.bold) c_token(colors.gray)"
      >
        Robert C. Martin
      </h4>
    </div>
    <div
      class="d_flex ai_center jc_flex-end mt_token(spacings.xxsmall)"
    >
      <div
        class="d_inline-flex fw_token(font.bold) h_3rem ai_center c_token(colors.white) p_0_token(spacings.xxsmall) bg-c_token(colors.secondary) bdr_token(border.radius) mr_calc(token(spacings.xxsmall)_/_2)"
      >
        R$Â 99,90
      </div>
    </div>
  </div>
</article>
`;
